src/tools/cryptoflex-tool.c

```diff
@@ -219,17 +231,44 @@ static int parse_public_key(const u8 *key, size_t keysize, RSA *rsa)
 	if (e == NULL)
 		return -1;
 	cf2bn(p, 4, e);
-	if (RSA_set0_key(rsa, n, e, NULL) != 1)
-	    return -1;
+	
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
+	if (!(rsa = RSA_new()) || 
+		!(pkey = EVP_PKEY_new()) || 
+		RSA_set0_key(rsa, n, e, NULL) != 1 || 
+		EVP_PKEY_assign_RSA(pkey, rsa)) {
```

Comment:
```suggestion
		EVP_PKEY_assign_RSA(pkey, rsa) != 1) {
```

Reply:
I don't believe `pkcs15-helper.c` and `pkcs15-helper.h` are ever used. No other source files include  `pkcs15-helper.h` and many of the defines in pkcs15-helper.h` are now in other OpenSC header files. 

They were initially introduced in c6954a9c333087034c94d3ef460487eecb90160c 2007-03-12 when support for compressed PIV certificates was added by Thomas Harning. The registration of piv objects, certificates, keys and pins were moved to the new module, but the module was setup to be usable by other emulated cards. but never used as far as I can tell. 

Then in 9946e237de1b09d4939448f354331a84b8e6f689 2007-06-21 says:

>  * The p15card-helper.c module is no longer used. The code to call the
       sc_pkcs15emu_* routines has been moved back into pkcs15-piv.c and uses
       existing OpenSC routines to parse the certificate to find the modulus_len.

I would delete these two files.

Reply:
Fixed.